src/common/webrtc/api.rs

```diff
@@ -426,6 +431,17 @@ where
             .get(0)
             .ok_or_else(|| WebRtcError::InvalidSDPOffer("no media description".to_owned()))?;
 
+        let remote_prio = attribute
+            .attribute("x-priority")
+            .flatten()
+            .map_or(Ok(u32::MAX), |a| a.parse::<u32>())
+            .unwrap_or(u32::MAX);
+        let local_prio = local_prio.unwrap_or(0);
+
+        if local_prio >= remote_prio {
```

Comment:
I think you could use `is_some_and` here to avoid needing to `unwrap_or` with a zero default into a shadowed `local_prio`.

```
if local_prio.is_some_and(|x| x >= remote_prio) {
    return Err(...);
```

Reply:
is_some_and is still unstable on rust 1.67 so we will have to push this until we reach 1.70 for micro-rdk